client/hosting/overview/components/migration-overview/components/migration-pending/use-cancel-migration.ts

```diff
@@ -0,0 +1,73 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useCallback, useEffect, useState } from 'react';
+import { useMigrationCancellation } from 'calypso/data/site-migration/landing/use-migration-cancellation';
+import { useSiteExcerptsQueryInvalidator } from 'calypso/data/sites/use-site-excerpts-query';
+import { useDispatch } from 'calypso/state';
+import { requestSite } from 'calypso/state/sites/actions';
+import type { SiteDetails } from '@automattic/data-stores';
+
+const useCancelMigration = ( site: SiteDetails ) => {
```

Comment:
With this, I extracted the cancellation logic, which is of considerable complexity.

I thought of also separating the `ConfirmModal` and `Notice` and control its state through context. But it would also introduce extra complexity just to extract 2 small components only used here so I gave up.